app/controllers/admin/facilities_controller.rb

```diff
@@ -118,6 +119,20 @@ def set_facility_group
     @facility_group = current_admin.accessible_facility_groups(:manage).friendly.find(params[:facility_group_id])
   end
 
+  def set_available_zones
+    zones =
+      if facility_region
+        facility_region.district.blocks
+      else
+        Region.where(type: RegionType.find_by_name("Block"))
+      end
+
+    @available_zones = zones.order(:name).pluck(:name)
+  end
+
+  def facility_region
+    @facility_region ||= Region.find_by(source_id: @facility.id) if @facility
```

Comment:
`if @facility.id`? The `new` action could have a blank `@facility` object with a `nil` ID.

I know it's not the case right now (since the creation happens in the controller action), but it's fairly common practice to extract that kind of stuff to a `before_action`, and I wouldn't want this to break if we do that.